django/db/models/fields/related.py

```diff
@@ -1967,7 +1967,7 @@ def set_attributes_from_rel(self):
         pass
 
     def value_from_object(self, obj):
-        return [] if obj.pk is None else list(getattr(obj, self.attname).all())
+        return [] if not obj._is_pk_set() else list(getattr(obj, self.attname).all())
```

Comment:
I imagined you wanted to minimize the diff but I think the following may be preferable:
```suggestion
        return list(getattr(obj, self.attname).all()) if obj._is_pk_set() else []
```